modules/gapi/test/common/gapi_parsers_tests_common.hpp

```diff
@@ -0,0 +1,384 @@
+        constexpr int minW = 16;
+        constexpr int minH = 16;
+        cv::Rect box;
+        box.width  = randInRange(minW, in_sz.width  + 2*border);
+        box.height = randInRange(minH, in_sz.height + 2*border);
+        box.x = randInRange(-border, in_sz.width  + border - box.width);
+        box.y = randInRange(-border, in_sz.height + border - box.height);
+        return box;
+    }
+
+    struct SSDitem
+    {
+        float image_id = 0.0f;
+        float label = 0.0f;
+        float confidence = 0.0f;
+        float rc_left = 0.0f;
+        float rc_top = 0.0f;
+        float rc_right = 0.0f;
+        float rc_bottom = 0.0f;
+    };
+
+    SSDitem generateItem(const int& i, const cv::Size& in_sz)
+    {
+        const auto normalize = [](int v, int range) { return static_cast<float>(v) / range; };
+
+        SSDitem it;
+        it.image_id = static_cast<float>(i);
+        it.label = static_cast<float>(randInRange(0, 9));
+        it.confidence = static_cast<float>(std::rand()) / RAND_MAX;
+        auto box = generateBox(in_sz);
+        it.rc_left   = normalize(box.x, in_sz.width);
+        it.rc_right  = normalize(box.x + box.width, in_sz.width);
+        it.rc_top    = normalize(box.y, in_sz.height);
+        it.rc_bottom = normalize(box.y + box.height, in_sz.height);
+
+        return it;
+    }
+
+    int EntryIndex(const int& side, const int& lcoords, const int& lclasses, const int& location, const int& entry)
+    {
+        int n = location / (side * side);
+        int loc = location % (side * side);
+        return n * side * side * (lcoords + lclasses + 1) + entry * side * side + loc;
+    }
+
+    struct DetectionObject
+    {
+        int xmin = 0, ymin = 0, xmax = 0, ymax = 0, class_id = 0;
+        float confidence = 0.0f;
+
+        DetectionObject(const double& x,
+                        const double& y,
+                        const double& h,
+                        const double& w,
+                        const int& class_idx,
+                        const float& conf,
+                        const double& h_scale,
+                        const double& w_scale)
+        {
+            this->xmin = static_cast<int>((x - w / 2) * w_scale);
+            this->ymin = static_cast<int>((y - h / 2) * h_scale);
+            this->xmax = static_cast<int>(this->xmin + w * w_scale);
+            this->ymax = static_cast<int>(this->ymin + h * h_scale);
+            this->class_id = class_idx;
+            this->confidence = conf;
+        }
+
+        bool operator <(const DetectionObject &s2) const
+        {
+            return this->confidence < s2.confidence;
+        }
+        bool operator >(const DetectionObject &s2) const
+        {
+            return this->confidence > s2.confidence;
+        }
+    };
+
+    struct YoloParams
+    {
+        int num = 5;
+        int coords = 4;
+        int classes = 80;
+    };
+
+    std::vector<DetectionObject> parse(const float* output_blob,
```

Comment:
`const float const` too?

Reply:
see if there are other instances of this

Reply:
No, send const float* to the function.